02-peer-protocol.md

```diff
@@ -430,6 +446,515 @@ would create a Denial of Service risk; therefore, forgetting it is recommended
+
+`channel_reserve_satoshi` has been omitted. The channel reserve is fixed at 1% of
+the total channel balance (sum of `funding_satoshis` from `open_channel2` and `accept_channel2`) 
+or the `dust_limit_satoshis`, whichever is greater.
+
+
+### The `accept_channel2` Message
+
+This message contains information about a node and indicates its
+acceptance of the new channel.
+
+1. type: 57 (`accept_channel2`)
+2. data:
+    * [`32*byte`:`temporary_channel_id`]
+    * [`u64`:`funding_satoshis`]
+    * [`u64`:`dust_limit_satoshis`]
+    * [`u64`:`max_htlc_value_in_flight_msat`]
+    * [`u64`:`htlc_minimum_msat`]
+    * [`u32`:`minimum_depth`]
+    * [`u16`:`to_self_delay`]
+    * [`u16`:`max_accepted_htlcs`]
+    * [`point`:`funding_pubkey`]
+    * [`point`:`revocation_basepoint`]
+    * [`point`:`payment_basepoint`]
+    * [`point`:`delayed_payment_basepoint`]
+    * [`point`:`htlc_basepoint`]
+    * [`point`:`first_per_commitment_point`]
+    * [`accept_tlvs`:`accept_tlv`]
+
+1. tlvs: `accept_tlvs`
+2. types:
+   1. type: 1 (`option_upfront_shutdown_script`)
+   2. data:
+       * [`u16`:`shutdown_len`]
+       * [`shutdown_len*byte`:`shutdown_scriptpubkey`]
+
+Rationale and Requirements are the same as listed above,
+for [`accept_channel`](#the-accept_channel-message) with the following additions.
+
+#### Requirements:
+
+The accepting node:
+    - MAY respond with a `funding_satoshis` value of zero.
+
+#### Rationale
+
+Accepter sends their `funding_satoshi` value here instead of allowing the opener to derive
+it from their `funding_compose` response so that the opener can notionally decide whether
+to complete the opening without exposing their output set.
+
+`channel_reserve_satoshi` has been omitted. The channel reserve is fixed at 1% of
+the total channel balance (sum of `funding_satoshis` from `open_channel2` and `accept_channel2`) 
+or the `dust_limit_satoshis`, whichever is greater.
+
+
+### Funding Composition
+There are three messages which nodes use to communicate their input and output set
+for a funding transaction: `funding_add_input`, `funding_add_output`, and `funding_add_complete`.
+
+To add inputs to a transaction, a node may send one or more `funding_add_input` messages.
+To add outputs to a transaction, a node may send one or more `funding_add_output` messages.
+
+Once a node has relayed all of their inputs and output contributions, it sends
+`funding_add_complete` to signal the completion of their contribution transmission.
+
+#### The `funding_add_input` Message
+
+This message contains information for inputs to a funding transaction.
+
+1. type: 58 (`funding_add_input`)
+2. data:
+    * [`32*byte`:`temporary_channel_id`]
+    * [`u16`:`num_inputs`]
+    * [`num_inputs*input_info`:`input_info`]
+
+1. subtype: `input_info`
+2. data:
+    * [`u64`:`sats`]
+    * [`sha256`:`prevtx_txid`]
+    * [`u32`:`prevtx_vout`]
+    * [`u16`:`prevtx_scriptpubkey_len`]
```

Comment:
I assume the reason for scriptpubkey inclusion is for bip 158 filters ?

Reply:
This hadn't occurred to me, but yes. I believe the reason for their inclusion was to allow for a quick rejection if the script is not obviously a segwit script.

Reply:
Okay IMO I find better to check in one place to avoid implementor assuming at the final check than scriptpubkey have been already validated (but not in a trusted way against blockchain view). Quick rejection will catch programming errors but not a malicious party.